lib/src/kbasesearchengine/main/IndexerWorker.java

```diff
@@ -841,18 +801,139 @@ private void indexObjectWrapperFn(final List<Object> input)
-            } catch (IOException e) {
-                throw new RetriableIndexingException(ErrorType.OTHER, e.getMessage(), e);
+
+        /** Note that an object may not be returned for a specified unresolvedGUID is the
+         * object parsing rules for that object are not found.
+         *
+         * @param objectRefPath
+         * @param unresolvedGUIDs
+         * @return
+         * @throws IndexingException
+         * @throws InterruptedException
+         * @throws RetriableIndexingException
+         */
+        private List<ObjectData> getObjectsByIds(final List<GUID> objectRefPath,
+                                                 final Set<GUID> unresolvedGUIDs)
+                throws IndexingException, InterruptedException, RetriableIndexingException {
+            final List<ObjectData> data = new ArrayList<>();
+
+            for(GUID unresolvedGUID: unresolvedGUIDs) {
+                // resolve reference
+                ResolvedReference resRef =
+                        resolveRefs(objectRefPath, ImmutableSet.of(unresolvedGUID)).
+                                                                     iterator().next();
+
+                // add subObjType and subObjId from the unresolved guid as it is
+                // not contained in the resolved reference
+                GUID guid = resRef.getResolvedReference();
+                guid = new GUID(guid.getStorageCode(),
+                                guid.getAccessGroupId(),
+                                guid.getAccessGroupObjectId(),
+                                guid.getVersion(),
+                                unresolvedGUID.getSubObjectType(),
+                                unresolvedGUID.getSubObjectId());
+                resRef = new ResolvedReference(resRef.getReference(),
+                                               guid,
+                                               resRef.getType(),
+                                               resRef.getTimestamp());
+
+                // get object from data source
+                final File tempFile;
+                try {
+                    FileUtil.getOrCreateSubDir(rootTempDir, guid.getStorageCode());
+                    tempFile = File.createTempFile("ws_srv_response_", ".json");
+                } catch (IOException ex) {
+                    throw new RetriableIndexingException(ErrorType.OTHER, ex.getMessage(), ex);
+                }
+
+                try {
+                    // make a copy to avoid mutating the caller's path
+                    final EventHandler handler = getEventHandler(guid);
+                    final SourceData obj = handler.load(
+                            Arrays.asList(resRef.getResolvedReference()), tempFile.toPath());
+
+                    final ObjectTypeParsingRules rules = getObjectTypeParsingRules(resRef);
+
+                    if(rules == null) {
+                        throw new FatalIndexingException(ErrorType.OTHER,
```

Comment:
This will shut the worker down - is that the desired behavior? UnprocessableEventException is an option if not.

Reply:
If it's not already documented, should note that a spec is required for any dependent objects that need to be processed because they're referenced by a parent.

Reply:
Changed to non-fatal exception

Reply:
added documentation